src/helpers.js

```diff
@@ -873,3 +873,36 @@ export function getIfaceSourceName(iface) {
 export function canDeleteDiskFile(disk) {
     return disk.type === "volume" || (disk.type === "file" && disk.source.file);
 }
+
+export function getStoragePoolPath(storagePools, poolName, connectionName) {
+    const pool = storagePools.find(pool => pool.name === poolName && pool.connectionName === connectionName);
+
+    return pool?.target?.path;
+}
+
+export function vmSupportsExternalSnapshots(config, vm, storagePools) {
+    const disks = Object.values(vm.disks);
+
+    // If at leat one disk has internal snapshot preference specified, use internal snapshot for all disk,
+    // as mixing internal and external is not allowed
+    if (disks.some(disk => disk.snapshot === "internal"))
+        return false;
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1280-20231030-133824-a492f2d5-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)